hw/dv/dpi/common/tcp_server/tcp_server.h

```diff
@@ -0,0 +1,95 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * Standard set of functions for setting up a TCP server
+ *
+ * This is intended to be used by simulation add-on DPI modules to provide
+ * basic TCP socket communication between a host and simulated peripherals.
+ */
+
+#ifndef TCP_SERVER_H_
+#define TCP_SERVER_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "tcp_buffer.h"
+
+#include <stdint.h>
+
+struct tcp_server_ctx {
+  // Written by requestor / read by socket
+  char *display_name;
+  uint16_t listen_port;
+  bool socket_run;
+  // Written by socket / read by requestor
+  tcp_buffer_struct *buf_in;
+  tcp_buffer_struct *buf_out;
+  int sfd;  // socket fd
+  int cfd;  // client fd
+};
+
+/**
+ * Start a TCP server
```

Comment:
Can you add a bit of a comment here on the type of socket that is opened? Especially the buffering behavior is good to explain, given how much time we've spent figuring that out.